src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/Apartment.java

```diff
@@ -466,19 +431,13 @@ public Apartment build() {
 				throw new IllegalStateException("The address of the apartment must be specified");
 			} else if (apartmentToBuild.getTitle() == null) {
 				throw new IllegalStateException("The title of the apartment must be specified");
-			} else if (apartmentToBuild.getTele() == null){
-				throw new IllegalStateException("You must specify if the apartment has WIFI");
-			} else if (apartmentToBuild.getWifi() == null){
-				throw new IllegalStateException("You must specify if the apartment has TV");
-			} else if (apartmentToBuild.getTerrace() == null){
-				throw new IllegalStateException("You must specify if the apartment has a Terrace");
-			} else if (apartmentToBuild.getTerrace().equals(Boolean.FALSE) && apartmentToBuild.getFloorAreaTerrace()>0){
+			} else if (!(apartmentToBuild.getTerrace()) && apartmentToBuild.getFloorAreaTerrace() > 0) {
```

Comment:
Il faut ajouter 3 attribut de classe boolean à builder correspondant à
`teleSet wifiSet terraceSet`
ces dernières sont initialisées à faux lors de la création du builder et passe à true lors de l'appel aux fonctions `Builder.setTele(boolean), Builder.setTerrace(boolean), Builder.setWifi(boolean)` .
La vérification qui a été retiré doit donc être restaurée et modifiée.

Reply:
Fixed in commit fb6b6931131540bc1ba62c798ee451d6adf93b88